src/modules/flow/netctl/netctl.c

```diff
@@ -0,0 +1,336 @@
+            sol_flow_send_error_packet(node, -r, "Could not disconnect to service: %s: %s",
+                     mdata->service_name, sol_util_strerrora(-r));
+    }
+
+    return 0;
+}
+
+static int
+connect_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    return connection_handle(node, data, true);
+}
+
+static int
+disconnect_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    return connection_handle(node, data, false);
+}
+
+static int
+get_service_state(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct network_service_data *mdata = data;
+    struct sol_netctl_service *service;
+    int r;
+
+    service = find_service_by_name(mdata->service_name);
+    if (!service) {
+        sol_flow_send_error_packet(node, ENOENT, "Did not found service name %s", mdata->service_name);
+        return 0;
+    }
+    r = sol_flow_send_string_packet(node,
+             SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__OUT,
+             sol_netctl_service_state_to_str(mdata->state));
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (mdata->state == SOL_NETCTL_SERVICE_STATE_ONLINE)
+            r = sol_flow_send_bool_packet(node,
+                         SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__ONLINE,
+                         true);
+    else
+            r = sol_flow_send_bool_packet(node,
+                         SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__ONLINE,
+                         false);
+
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+static int
+set_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct network_service_data *mdata = data;
+    const char *in_value;
+    int r;
+
+    r = sol_flow_packet_get_string(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    mdata->service_name = strdup(in_value);
+    SOL_NULL_CHECK(mdata->service_name, -ENOMEM);
+
+    return 0;
+}
+
+static int
+open_network_service(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    struct network_service_data *mdata = data;
+    const struct sol_flow_node_type_netctl_service_options *opts;
+    int r;
+
+    SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK(options,
+        SOL_FLOW_NODE_TYPE_NETCTL_SERVICE_OPTIONS_API_VERSION,
+        -EINVAL);
+    opts = (const struct sol_flow_node_type_netctl_service_options *)
+        options;
+
+    if (opts->name) {
+        mdata->service_name = strdup(opts->name);
+        SOL_NULL_CHECK(mdata->service_name, -ENOMEM);
+    }
+
+    mdata->state = SOL_NETCTL_SERVICE_STATE_UNKNOWN;
+
+    r = sol_netctl_add_service_monitor(service_status_cb, node);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_netctl_add_error_monitor(error_cb, node);
+    SOL_INT_CHECK(r, < 0, r);
+
```

Comment:
@barbieri That's what I am concern. Because all the services got in FBP are from sol_netctl_add_service_monitor callback network_service_cb. 
So we create netctl/service node and then connect to given service name. Maybe we cannot successfully connect because service_list by calling API sol_netctl_find_service_name is not ready yet for asynchronous delay reason.
I think the best time to connect service is after emitting status packet first time in service_status_cb.
Initial Tick maybe still is not time for user to connect service, just delay some time for user to connect service.
Maybe I didn't totally catch your meaning. I saw the examples, this dispatch just send bool packet in the platform_open which still cannot ensure service has been catched in our network_service_cb case.